### ISSUE TITLE
Page Performance reports don't show data (empty graph) when viewing performance for a flattened page URL

### ISSUE THREAD

**Issue by mattab** [piwik/piwik#17807]
Page Performance reports work well in the default "Pages" report's hierarchical view,
but as soon as the report is flattened (a very often used feature),
then Page performance reports don't work (empty graph).

## Expected Behavior

Graph with data eg:

![Screenshot from 2021-07-23 16-19-50](https://user-images.githubusercontent.com/466765/126736962-51f4b207-8d93-441c-9099-4e177cf954a1.png)

## Current Behavior

![Screenshot from 2021-07-23 16-17-34](https://user-images.githubusercontent.com/466765/126736842-5d687f15-658b-440b-95a6-6d659c787df1.png)

## Steps to Reproduce (for Bugs)
* Go to Behaviour > Pages report.
* Flatten the report
* Hover on a row and click "open page performance report"
* The report displays no data

Note: on the demo it does not anyway as it's missing performance data. But can be reproduce in any other matomo

## Your Environment

* Matomo Version: 4.3.x

**Comment by Jaimeapo** [comment]
Hi, We are using Matomo 4.4.1 and we are experiencing missing data on some installations.
I see an error: 204 No Response
![image](https://user-images.githubusercontent.com/2892389/136794606-82bb81d1-7464-43e2-8f90-8aea8d9658bd.png)

And the **Query String Parameters** the parameters has been shown as well, but the report doesn't show anything.
```
fa_pv: 1
fa_fp[0][fa_vid]: xVL8re
fa_fp[0][fa_fv]: 1
pf_net: 65
pf_srv: 97
pf_tfr: 70
pf_dm1: 1872
```

![image](https://user-images.githubusercontent.com/2892389/136795705-fed1793b-db33-4a9b-92a1-c2362d81c9ed.png)

**Comment by tsteur** [comment]
@Jaimeapo the 204 response is actually expected for all tracking requests. Generally this sounds like a different problem to the originally reported issue. 

As there seems to be a premium feature involved (form analytics), could you send this information that you mentioned to shop@innocraft.com ?

**Comment by Stan-vw** [comment]
We tried to reproduce this bug. We found some page performance charts don't have data (as indicated by the ticket) but many others do have data.
We played around with flattened reports and unflattened reports. It seems that the page performance graphs are only empty for rows that:
- don't exist in the unflattened report, namely `/`
- aren't links in the unflattened report

We'll need to investigate a bit further when this chart doesn't show data, and if it should in those cases.

**Comment by tsteur** [comment]
fyi this can be reproduced in our own analytics account

**Comment by BloodyIron** [comment]
I'm having this issue, can we get this addressed please? I observed it in #21811

**Comment by sgiehl** [comment]
@BloodyIron This will be fixed with Matomo 5.0.2
